src/iconv.jl

```diff
@@ -1,20 +1,48 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
-import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
-export StringEncoder, StringDecoder, encode, decode
+import Base: close, eof, flush, read, readall, write, show
+import Base.Libc: errno, E2BIG, EINVAL, EILSEQ
+export StringEncoder, StringDecoder, encode, decode, StringErr, StringError
+
+module StringErr
```

Comment:
Instead of an enum, I wonder whether we should use specific exception types? I think we should ask for guidance about what's the best solution e.g. on julia-users. That's mostly a matter of convention, the idea being that users must be able to check easily for a specific exception, and avoid catching unexpected ones. Exceptions could have specific fields allowing to retrieve the relevant information (e.g. the invalid sequence).